batchflow/tests/segmentation_metrics_by_pixels_test.py

```diff
@@ -0,0 +1,312 @@
+                'dor' : ['diagnostics_odds_ratio'],
+                'fos' : ['f1_score', 'dice'],
+                'jac' : ['jaccard', 'iou']}
+# List all metrics' names from METRICS_DICT values.
+METRICS_LIST = list(chain.from_iterable(METRICS_DICT.values()))
+
+BATCH_SIZE = 2
+IMAGE_SIZE = 2
+NUM_CLASSES = 3
+
+# Set targets.
+TARGETS = np.array([0, 1, 2, 2, 0, 0, 1, 1]).reshape(BATCH_SIZE, IMAGE_SIZE, IMAGE_SIZE)
+# Set predictions as 'labels'.
+LABELS = np.array([0, 1, 1, 0, 2, 0, 1, 1]).reshape(BATCH_SIZE, IMAGE_SIZE, IMAGE_SIZE)
+# Onehots are basically like probas, just with all 0 and a single 1.
+PROBA = np.eye(NUM_CLASSES)[LABELS]
+# Logit function gives ±infs on degenerate case of 0s and 1s, but it's okay for sigmoid function.
+LOGITS = np.log(PROBA / (1. - PROBA))
+# First param stands for predictions variable, second — for predictions type, third — for axis with class info.
+PREDICTIONS = [(LABELS, 'labels', None),
+               (PROBA, 'proba', 3),
+               (LOGITS, 'logits', 3)]
+
+class TestShape:
+    """
+    This class checks the shape of metrics' return value for all combinations of both aggregation types.
+
+    There is a following pattern in both tests:
+    0. Each function is preceded by data for it's parametrization.
+    1. Fixtures with preset parameters are applied.
+    2. Instance of SegmentationMetricsByPixels is being created.
+    3. Metric is being evaluated with given parameters.
+    4. It's result's shape is being compared with expected one.
+    """
+
+    # First param stands for batch aggregation, second — for multiclass one, third represents expected output shape.
+    params = [(None, None, (BATCH_SIZE, NUM_CLASSES)),
+              (None, 'micro', (BATCH_SIZE,)),
+              (None, 'macro', (BATCH_SIZE,)),
+              ('mean', None, (NUM_CLASSES,)),
+              ('mean', 'micro', None),
+              ('mean', 'macro', None)]
+
+    @pytest.mark.parametrize('predictions, fmt, axis', PREDICTIONS)
+    @pytest.mark.parametrize('batch_agg, multi_agg, exp_shape', params)
+    def test_shape(self, predictions, fmt, axis, batch_agg, multi_agg, exp_shape):
+        """
+        Function compares expected return value shape with actual return value shape
+        of metric evaluation with given params for all metrics from METRICS_LIST.
+
+        Parameters
+        ----------
+        predictions : np.array
+            Variable name containing predictions' array of desired format
+
+        fmt : string
+            Denotes predictions format
+
+        axis : None or int
+            A class axis
+
+        batch_agg : string
+            Cross-batch aggregation type
+
+        multi_agg : string
+            Multiclass agregation type
+
+        exp_shape : None or tuple
+            Expected return value shape
+        """
+        metric = SegmentationMetricsByPixels(TARGETS, predictions, fmt, NUM_CLASSES, axis)
+        for metric_name in METRICS_LIST:
```

Comment:
If the test fails before assert (e.g. on metric evaluation) it would be quite hard to find on which metric exactly did it fail. You can make another parameter `metric_name` and get rid of loop inside the function.